deepchecks/tabular/checks/integrity/outlier_sample_detection.py

```diff
@@ -42,71 +49,84 @@ class OutlierSampleDetection(SingleDatasetCheck):
     """
 
     def __init__(
             self,
             columns: Union[Hashable, List[Hashable], None] = None,
             ignore_columns: Union[Hashable, List[Hashable], None] = None,
-            num_nearest_neighbors: int = 10,
+            nearest_neighbors_percent: float = 0.01,
             extent_parameter: int = 3,
-            n_samples: int = 50_000,
+            n_samples: int = 5_000,
             n_to_show: int = 5,
             random_state: int = 42,
+            timeout: int = 10,
             **kwargs
     ):
         super().__init__(**kwargs)
         if not isinstance(extent_parameter, int) or extent_parameter <= 0:
             raise DeepchecksValueError('extend_parameter must be a positive integer')
-        if not isinstance(num_nearest_neighbors, int) or num_nearest_neighbors <= 0:
-            raise DeepchecksValueError('num_nearest_neighbors must be a positive integer')
+        if nearest_neighbors_percent <= 0 or nearest_neighbors_percent > 1:
+            raise DeepchecksValueError('nearest_neighbors_percent must be a float between 0 and 1')
         self.columns = columns
         self.ignore_columns = ignore_columns
-        self.num_nearest_neighbors = num_nearest_neighbors
+        self.nearest_neighbors_percent = nearest_neighbors_percent
         self.extent_parameter = extent_parameter
         self.n_samples = n_samples
         self.n_to_show = n_to_show
         self.random_state = random_state
+        self.timeout = timeout
 
     def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResult:
         """Run check."""
+        timeout_msg = f'Check has reached specified timeout of {format_number(self.timeout)} seconds, try reducing ' \
+                      f'n_samples, select a subset of columns or increase the timeout. '
+        signal.signal(signal.SIGALRM, get_time_out_handler(timeout_msg))
+        signal.alarm(self.timeout)
         if dataset_type == 'train':
             dataset = context.train
         else:
             dataset = context.test
         dataset = dataset.sample(self.n_samples, random_state=self.random_state, drop_na_label=True)
         df = select_from_dataframe(dataset.data, self.columns, self.ignore_columns)
-        if self.num_nearest_neighbors >= len(df):
-            logger.warning('Passed num_nearest_neighbors %s which is greater than the number of samples in the dataset'
-                           , self.num_nearest_neighbors)
-            self.num_nearest_neighbors = len(df) - 1
-
-        # Calculate distances matrix and retrieve nearest neighbors based on distance matrix.
-        df_cols_for_gower = df[dataset.cat_features + dataset.numerical_features]
-        is_categorical_arr = np.array(df_cols_for_gower.columns.map(lambda x: x in dataset.cat_features), dtype=bool)
+        num_neighbours = max(int(self.nearest_neighbors_percent * df.shape[0]), 5)
```

Comment:
Have this magic number as an upper case constant in the top of the py file
MIN_NUM_NEIGHBOURS